cloudevents/working-drafts/xml-format.md

```diff
@@ -0,0 +1,277 @@
+# XML Event Format for CloudEvents
+
+## Abstract
+
+This format specification for CloudEvents defines how events are expressed
+in XML documents.
+
+## Status of this document
+
+This document is a working draft.
+
+## Table of Contents
+
+1. [Introduction](#1-introduction)
+2. [Attributes](#2-attributes)
+3. [Data](#3-data)
+4. [Envelope](#4-envelope)
+5. [XML Batch Format](#5-xml-batch-format)
+6. [Examples](#6-examples)
+
+## 1. Introduction
+
+[CloudEvents][ce-spec] is a standardized and protocol-agnostic definition of the
+structure and metadata description of events. This specification defines how
+elements defined in the CloudEvents specification are to be represented using
+[Extensible Markup Language (XML)](xml-spec) documents.
+
+* The [Attributes](#2-attributes) section describes the representation and
+data type mappings for CloudEvents context attributes.
+
+* The [Data](#3-data) section defines the container for the data portion of a
+CloudEvent.
+
+* The [Envelope](#4-envelope) section defines an XML element for CloudEvents
+attributes and an associated media type.
+
+* The [Batch](#5-xml-batch-format) section describes how multiple CloudEvents
+can be packaged into a single XML document.
+
+### 1.1. Conformance
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC2119][rfc2119].
+
+### 1.2 Approach
+
+This XML representation used is deliberately verbose in favor of readability
+and deterministic processing.
+
+Preservation of type information for CloudEvent attributes is supported allowing
+custom extensions to be communicated without type loss.
+
+A schema-less approach has been taken favoring convention over rigid document
+structure.
+
+The namespace `http://cloudevents.io/xmlformat/v1` SHOULD be used, when
+namespace prefixes are used a prefix of `ce` is preferred but MUST NOT be expected
+from an XML document processing perspective.
+
+An XML document preamble SHOULD be included to ensure deterministic processing.
+
+## 2. Attributes
```

Comment:
Just in terms of clarity, I wonder whether we should put the envelope first. It feels a little odd to start with attributes without knowing what they're going to be part of.
Only something to consider, and can definitely be ignored.

Reply:
I basically followed the same structure as the JSON format ... so I get your point, but do we wave the "consistency flag" here ?

Reply:
Ah, interesting. In the JSON format there isn't really anything "to" the envelope - it's just an object. I'm happy for you to stick with this as it is. We could always re-evaluate wholesale for a later release.